src/web/components/App.tsx

```diff
@@ -351,6 +351,54 @@ export const App = ({ CAPI, NAV }: Props) => {
                     />
                 </Hydrate>
             ))}
+            {CAPI.guideAtoms.map((guideAtom) => (
+                <Hydrate root="guide-atom" index={guideAtom.guideIndex}>
+                    <GuideAtom
+                        id={guideAtom.id}
+                        title={guideAtom.title}
+                        html={guideAtom.html}
+                        image={guideAtom.img}
+                        credit={guideAtom.credit}
+                        pillar={pillar}
+                        likeHandler={() => {
+                            const componentEvent: OphanComponentEvent = {
+                                component: {
+                                    componentType: 'GUIDE_ATOM',
+                                    id: guideAtom.id,
+                                    labels: [],
+                                    products: [],
+                                },
+                                action: 'LIKE',
+                            };
+                            submitComponentEvent(componentEvent);
+                        }}
```

Comment:
Maybe this is another good opportunity to make this clearer and easier to reuse?

```
// Function that returns a function
const componentEventHandler = (componentType, id, action) => () => {
                            const componentEvent: OphanComponentEvent = {
                                component: {
                                    componentType,
                                    id,
                                    labels: [],
                                    products: [],
                                },
                                action,
                            };
                            submitComponentEvent(componentEvent);
                        }

```
```suggestion
	likeHandler={componentEventHandler('GUIDE_ATOM', guideAtom.id, 'LIKE')}
```

and you can reuse for this and other atoms:

`dislikeHandle={componentEventHandler('GUIDE_ATOM', guideAtom.id, 'DISLIKE')}`

What do you reckon?

Reply:
Yeah, that looks like a plan